generator/src/main/java/com/scottlogic/deg/generator/utils/FileUtils.java

```diff
@@ -85,4 +88,20 @@ public File getTraceFile(GenerationConfigSource configSource) {
         String filenameWithoutExtension = configSource.getOutputPath().toString().replaceAll("\\.[^.]+$", "");
         return Paths.get(filenameWithoutExtension + "-trace.json").toFile();
     }
+
+    /**
+     * we wrap Files.createDirectories() so that we can mock it out in unit tests.
+     *
+     * @param dir the directory we want to create
+     * @return true is the directory was created successfully, false if the parent directory already exists
+     * @throws IOException if we are unable to create the directory due to an I/O error
+     */
+    public boolean createDirectories(Path dir) throws IOException {
```

Comment:
instead of try catch, should we do 
if (Files.notExists(dir)) { Files.createDirectories(dir); }

Reply:
alternatively

is there a reason to not also catch IOException here as well as fileAlreadyExists

Reply:
the reason for catching `FileAlreadyExistsException` whilst creating the parent directory is that it is only thrown when the parent directory is a pre-existing `File` e.g. the user has asked to send output to a file `/tmp/README/output.csv` where `/tmp` is a directory `/tmp/README` is a pre-existing file.
IOException is thrown if we are unable to create the directory _due to an I/O error_ (according to the javadoc)